site/src/sphinx/client-decorator.rst

```diff
@@ -67,6 +67,116 @@ If your decorator is expected to be reusable, it is recommended to define a new
     // Using a lambda expression:
     cb.decorator(delegate -> new AuditClient(delegate));
 
+The order of decoration
+-----------------------
+
+The decorators are executed in reverse order of the insertion. The following example shows which order
+the decorators are executed by printing the messages.
+
+.. code-block:: java
+
+    import com.linecorp.armeria.client.WebClient;
+
+    ClientBuilder cb = Clients.builder(...);
+
+    // #2 decorator
+    cb.decorator((delegate, ctx, req) -> {
+        System.err.println("Secondly, executed.");
+        ...
+    });
+
+    // #1 decorator.
+    // No matter decorator() or option() is used, decorators are executed in reverse order of the insertion.
+    cb.option(ClientOption.DECORATION, ClientDecoration.of((delegate, ctx, req) -> {
```

Comment:
I found `ClientDecoration.of(...)` does not accept `DecoratingHttpClientFunction`.
https://github.com/line/armeria/blob/6f21290a3e6e3bb2af22d8a9061d472dcabda545/core/src/main/java/com/linecorp/armeria/client/ClientDecoration.java#L43

Reply:
And how about making the method body complete for the detail example?
```java
// #2 decorator
cb.decorator((delegate, ctx, req) -> {
    System.err.println("Secondly, executed.");
    final HttpResponse res = delegate.execute(ctx, req);
    System.err.println("Thirdly, executed.");
    return res;
});

// #2 decorator
cb.option(ClientOption.DECORATION, ClientDecoration.of(delegate -> (ctx, req) -> {
    System.err.println("Firstly, executed");
    final HttpResponse res = delegate.execute(ctx, req);
    System.err.println("Fourthly, executed.");
    return res;
}));
```

Reply:
> I found ClientDecoration.of(...) does not accept DecoratingHttpClientFunction.

Oops, you've already added it to this PR 😅

Reply:
Had chat with @minwoox, we are going to focus on the exeution orders.
If any questions or requests on this feature, we will improve our documentation. :-)